api/scpca_portal/models/computed_file.py

```diff
@@ -93,19 +104,38 @@ def get_portal_metadata_file(cls, projects, download_config: Dict) -> Self:
         computes a zip archive with metadata and readme files, and
         creates a ComputedFile object which it then saves to the db.
         """
-        with ZipFile(common.OUTPUT_PORTAL_METADATA_ZIP_FILE_PATH, "w") as zip_file:
+        libraries = Library.objects.all()
+        # If the query returns empty, then an error occurred, and we should abort early
+        if not libraries.exists():
+            return
+
+        libraries_metadata = utils.filter_dict_list_by_keys(
+            [lib for library in libraries for lib in library.get_combined_library_metadata()],
+            common.METADATA_COLUMN_SORT_ORDER,
+        )
+
+        zip_file_path = cls.get_local_file_path(download_config)
+        with ZipFile(zip_file_path, "w") as zip_file:
+            # Readme file
             zip_file.writestr(
                 readme_file.OUTPUT_NAME,
                 readme_file.get_file_contents(
                     download_config,
                     projects,
                 ),
             )
+            # Metadata file
+            zip_file.writestr(
+                metadata_file.get_file_name(download_config),
+                metadata_file.get_file_contents(libraries_metadata),
+            )
+
         computed_file = cls(
             portal_metadata_only=True,
             s3_bucket=settings.AWS_S3_BUCKET_NAME,
+            # Would it be better to create this as a class attribute instead of common property?
```

Comment:
```suggestion
```

This makes sense to keep in common since it is a special case output file that does not belong to any specific project or sample.

Reply:
See at 651d753c49c931eb71237155e64ad99226d6e9c8.